### PR TITLE
Feature/replication ui

### PR DESCRIPTION
**Pull request by hkernbach**
This PR introduces a new replication UI. 

A new view and menu entry will appear if you are logged in into the `_system` database:
`Replication`

This view will change its content depending on what replication setup is running (mode):
1. No replication running
2. Master/Slave (default single applier - per database)
3. Master/Slave (global replier - per server)
4. Active Failover

### CODE REVIEW

**Comment by hkernbach** [comment]
Not edited CHANGELOG.md by now, because it is not clear whether this feature will make it into 3.3.9 or 3.3.10. Keep In mind!

Also after merge we need a new PR for devel.

File: arangod/VocBase/vocbase.cpp [hunk 1]
```diff
@@ -1654,15 +1658,15 @@ void TRI_vocbase_t::updateReplicationClient(TRI_server_id_t serverId,
...
 TRI_vocbase_t::getReplicationClients() {
-  std::vector<std::tuple<TRI_server_id_t, double, TRI_voc_tick_t>> result;
+  std::vector<std::tuple<TRI_server_id_t, double, double, TRI_voc_tick_t>> result;
 
   READ_LOCKER(readLocker, _replicationClientsLock);
 
   for (auto& it : _replicationClients) {
     result.emplace_back(
-        std::make_tuple(it.first, it.second.first, it.second.second));
+        std::make_tuple(std::get<0>(it), std::get<0>(std::get<1>(it)), std::get<1>(std::get<1>(it)), std::get<2>(std::get<1>(it)))); // TODO
```
[the remark this answers is not in the document]
Comment (hkernbach) [reply arangod/VocBase/vocbase.cpp]: just forgotten. TODO statement can be deleted.

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 2]
```diff
@@ -1932,6 +1938,37 @@ void RestReplicationHandler::handleCommandApplierGetState() {
...
+  if (_request->databaseName() != StaticStrings::SystemDatabase) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_FORBIDDEN,
+                  "global inventory can only be fetched from within _system database");
+  }
+  DatabaseFeature* databaseFeature = application_features::ApplicationServer::getFeature<DatabaseFeature>("Database");
+
+  VPackBuilder builder;
+  builder.openObject();
+  for (auto& name : databaseFeature->getDatabaseNames()) {
+    VPackBuilder tmpBuilder;
```
Comment (jsteemann) [inline arangod/RestHandler/RestReplicationHandler.cpp]: No need for a temporary `tmpBuilder` here. Please use the existing `builder`!

File: arangod/V8Server/v8-replication.cpp [hunk 3]
```diff
@@ -600,11 +601,49 @@ static void StateApplierReplication(v8::FunctionCallbackInfo<v8::Value> const& a
...
+  if (args.Length() != 0) {
+    TRI_V8_THROW_EXCEPTION_USAGE("stateAll()");
+  }
+
+  DatabaseFeature* databaseFeature = application_features::ApplicationServer::getFeature<DatabaseFeature>("Database");
+
+  VPackBuilder builder;
+  builder.openObject();
+  for (auto& name : databaseFeature->getDatabaseNames()) {
+    VPackBuilder tmpBuilder;
```
Comment (jsteemann) [inline arangod/V8Server/v8-replication.cpp]: Please reuse `builder` here.

File: arangod/VocBase/vocbase.cpp (hunk 1 above)
Comment (jsteemann) [inline arangod/VocBase/vocbase.cpp]: this is hardly readable.
can we please use `it.first` and `it.second` here?

File: js/apps/system/_admin/aardvark/APP/frontend/js/views/applierView.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        url: arangoHelper.databaseUrl('/_admin/aardvark/replication/mode'),
+        contentType: 'application/json',
+        success: function (data) {
+          if (data.mode === 3) {
+            self.getApplierState(this.endpoint, true);
+          } else if (data.mode === 2) {
+            self.getApplierState(this.endpoint, true);
+          } else {
+            self.getApplierState(this.endpoint, false);
+          }
```
Comment (jsteemann) [inline js/apps/system/_admin/aardvark/APP/frontend/js/views/applierView.js]: this `if ... else if ... else` can be replaced with just `self.getApplierState(this.endpoint, data.mode === 3 || data.mode === 2);`

Comment (jsteemann) [reply js/apps/system/_admin/aardvark/APP/frontend/js/views/applierView.js]: or at least the code for modes 2 and 3 can be put into a single if branch

**Comment by jsteemann** [comment]
Do we also need to add the `stateAll` function to the applier in js/client/modules/@arangodb/replication.js? Otherwise the function will not be present in arangosh, but only in arangod.
For consistency I think it should be added in arangosh too.

**Pull request merged**